analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -18,11 +18,25 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Collections.Immutable;
+using StyleCop.Analyzers.Lightup;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn
 {
     public sealed class ProgramState
     {
-        public static readonly ProgramState Empty = new ProgramState();
-        // ToDo: Some stuff is missing here
+        public static readonly ProgramState Empty = new ProgramState(ImmutableDictionary<IOperationWrapperSonar, SymbolicValue>.Empty);
+
+        private readonly ImmutableDictionary<IOperationWrapperSonar, SymbolicValue> operationValue;
+
+        public SymbolicValue this[IOperationWrapperSonar operation] => operationValue.TryGetValue(operation, out var value) ? value : null;
```

Comment:
nitpick: public properties/accessor before private fields?

Reply:
This is correct. We should have all fields (public, protected, private) followed by all properties (public, protected, private).